src/os/bluestore/NVMEDevice.cc

```diff
@@ -919,7 +916,18 @@ int NVMEDevice::open(const string& p)
 
   driver->register_device(this);
   block_size = driver->get_block_size();
-  size = driver->get_size();
+
+  str = index(buf, ':');
+  offset = strtoll(str + 1, NULL, 16);
+  str = rindex(buf, ':');
+  size = strtoll(str + 1, NULL, 16);
```

Comment:
will size be assigned in? In the legacy_config_opt.h, you describe:

> "defintely the size is determined by the bluestore_block_size."

Reply:
Yes,  size is assigned with the passed value. Open is generic,  it could be open on db, wal file, and the size is passed fro the calling function, and the calling function passed the value set in the conf file.

Reply:
so i mean line 922 and 923 should be removed?